### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by arsinclair** [TypeStrong/fork-ts-checker-webpack-plugin#680]
## Current behavior
I recently have done several package upgrades in my project, including webpack, but _fork-ts-checker_ wasn't upgraded. The project uses Electron.
I noticed that after the upgrade, when compilation is started there's a huge CPU spike and the computer freezes for about a minute, before it can operate again. Even the mouse pointer freezes. Disabling _ForkTsCheckerWebpackPlugin_ plugin in webpack config resolves the issue. It happens right after the renderer code has been compiled.

I tried passing logger into the options of the plugin, but it doesn't provide any valuable insight of what's happening.

## Expected behavior
Shouldn't freeze.

## Steps to reproduce the issue
I don't see any particular reproduction steps, as the issue manifested after one of the package upgrades (there was no code change).

## Issue reproduction repository
Unfortunately, my organisation doesn't allow sharing the repository, but I'm happy to run any troubleshooting steps that are necessary to pinpoint the issue.

## Environment
 - **fork-ts-checker-webpack-plugin**: 6.4.2
 - **typescript**: 4.5.2
 - **eslint**: 8.3.0
 - **webpack**: 5.64.4
 - **os**: "Debian GNU/Linux 10 (buster)"

**Comment by rafaeelaudibert** [comment]
I'd +1 this. After a recent upgrade in our containers, when this plugin starts executing, it uses 100% CPU for ~1 minute.

**Comment by arsinclair** [comment]
I rolled back all the changes, but this problem didn't go away. I guess it is not related to the upgrade, after all. I noticed that on Node v14 it is actually much worse than on Node v16. It completely froze my computer, and didn't unfreeze it after 15 minutes.

What I noticed is that disk activity LED on my tower is showing constant activity when this happens.

I tried to run this together with _top_ and _iotop_. Top shows that most CPU activity is spent on `kswapd0`, and disk activity didn't show anything helpful, but because the computer becomes very very slow at this moment, maybe these tools just didn't have enough processor time to accumulate the data.

I also had a lot of apps running in the background, primarily electron apps - Discord, Spotify, Brave browser, etc. After I have closed all the apps, the freeze has decreased significantly.
